src/test/java/seedu/address/model/task/DeadlineTaskTest.java

```diff
@@ -43,9 +43,12 @@ public void equals_isEqual_returnsTrue() throws Exception {
 
     @Test
     public void equals_notEqual_returnsFalse() throws Exception {
-        final DeadlineTask other = new DeadlineTask(new Name("Deadline Task Name"), UNIX_EPOCH.plusDays(2));
-        assertFalse(deadlineTask.equals(other));
-        assertFalse(deadlineTask.hashCode() == other.hashCode());
+        final DeadlineTask other1 = new DeadlineTask(new Name("Deadline Task Name"), UNIX_EPOCH.plusDays(2), false);
+        final DeadlineTask other2 = new DeadlineTask(new Name("Deadline Task Name"), UNIX_EPOCH.plusDays(1), true);
+        assertFalse(deadlineTask.equals(other1));
+        assertFalse(deadlineTask.hashCode() == other1.hashCode());
+        assertFalse(deadlineTask.equals(other2));
+        assertFalse(deadlineTask.hashCode() == other2.hashCode());
```

Comment:
Ugh I haven't taken a close look at this PR yet, sorry. (Too busy working on the new parser, which must must must be done by this cycle)

If you need this in ASAP, maybe get another reviewer?